src/test/java/htsjdk/samtools/CRAMComplianceTest.java

```diff
@@ -362,6 +371,81 @@ public void testBAMThroughCRAMRoundTripViaPath() throws IOException {
 
+    @DataProvider(name="obsoleteDataseriesCRAM")
+    public Object[][] obsoleteDataseriesCRAM() {
+        return new Object[][] {
+
+                // 1301_slice_aux.cram has legacy dataseries TC, TN present in it.
+                // TC, TN are now removed and are no longer written by htsjdk
+                // This file is used to test if CRAM reader can ignore these dataseries.
+                // 1301_slice_aux.cram is taken from hts-specs repo. It uses reference files, ce.fa and ce.fa.fai.
+                // 1301_slice_aux.cram file: https://github.com/samtools/hts-specs/blob/master/test/cram/3.0/passed/1301_slice_aux.cram
+                // Reference files: https://github.com/samtools/hts-specs/tree/master/test/cram
+                { new File(TEST_DATA_DIR, "1301_slice_aux.cram"),
+                        new File(TEST_DATA_DIR, "ce.fa") },
+        };
+    }
+
+    @Test(dataProvider = "obsoleteDataseriesCRAM", description = "round trip a CRAM file with obsolete dataseries TC and TN. TC and TN should not be present in the output CRAM file")
+    public void testRoundtripObsoleteDataseries(final File sourceFile, final File referenceFile) throws IOException {
+        final CRAMEncodingStrategy testStrategy = new CRAMEncodingStrategy();
+        final File tempOutCRAM = File.createTempFile("testRoundTrip", ".cram");
+        tempOutCRAM.deleteOnExit();
+        CRAMTestUtils.writeToCRAMWithEncodingStrategy(testStrategy, sourceFile, tempOutCRAM, referenceFile);
+        assertRoundTripFidelity(sourceFile, tempOutCRAM, referenceFile, false);
+        try (InputStream cramInputStream = new BufferedInputStream(new FileInputStream(tempOutCRAM));
+            CountingInputStream countingInputStream = new CountingInputStream(cramInputStream);
+            CramContainerIterator containerIterator = new CramContainerIterator(countingInputStream)) {
+            CramHeader cramHeader = containerIterator.getCramHeader(); //actual cram header
+            final long containerByteOffset = countingInputStream.getCount();
+            do {
+
+                // read the container and its CompressionHeader
+                Container container = new Container(cramHeader.getCRAMVersion(), countingInputStream, containerByteOffset);
+                CompressionHeader compressionHeader = container.getCompressionHeader();
+
+                // get the CompressionHeaderEncodingMap
+                CompressionHeaderEncodingMap encodingMap = compressionHeader.getEncodingMap();
+
+                //make sure obsolete DataSeries TC, TN are not present in the CompressionHeaderEncodingMap
+                for (final DataSeries dataseries : CompressionHeaderEncodingMap.DATASERIES_NOT_READ_BY_HTSJDK) {
+                    Assert.assertNull(encodingMap.getEncodingDescriptorForDataSeries(dataseries),
+                            "Unexpected encoding key found: " + dataseries.getCanonicalName());
+                }
+            } while(containerIterator.hasNext());
```

Comment:
You can iterate through the containers directly by using the container iterator that you already have (and skip the counting input stream, the byte offsets, and the cram header) by simplifying this loop to:

```suggestion
        try (final InputStream cramInputStream = new BufferedInputStream(new FileInputStream(tempOutCRAM));
            final CramContainerIterator containerIterator = new CramContainerIterator(cramInputStream)) {
            while (containerIterator.hasNext()) {
                // read the container and its CompressionHeader
                final Container container = containerIterator.next();
                final CompressionHeader compressionHeader = container.getCompressionHeader();

                // get the CompressionHeaderEncodingMap
                final CompressionHeaderEncodingMap encodingMap = compressionHeader.getEncodingMap();

                //make sure obsolete DataSeries TC, TN are not present in the CompressionHeaderEncodingMap
                for (final DataSeries dataseries : CompressionHeaderEncodingMap.DATASERIES_NOT_READ_BY_HTSJDK) {
                    Assert.assertNull(encodingMap.getEncodingDescriptorForDataSeries(dataseries),
                            "Unexpected encoding key found: " + dataseries.getCanonicalName());
                }
            }
        }```